### PR TITLE
Remove unit benchmarks

### LINKED ISSUE #5475: Consider removing some unit test style benchmarks
Benchmark run time is too long for these sorts of benchmarks. Plus they run very quickly, so are more vulnerable to noise. We'll get better coverage if all benchmarks are integration-style, like something a user would actually be trying to do. We lose a bit of granular information without unit tests, but benchmarks always need some human investigation anyway so that's not a big loss.

### PR DESCRIPTION
**Pull request by trexfeathers**
### Description

Closes #5475 

- Introduces `iris.tests.stock.realistic_4d_w_everything()` - a `Cube` with as many features as possible - aimed at exercising as much of Iris as possible, useful for wide coverage benchmarks.
- Deactivates some out-dated benchmarks from being run by default.
- Replaces lost coverage from the above retirement with benchmarks that use the new realistic 'everything `Cube`'.
- Changed README and docstring advice to no longer encourage unit-style testing and instead encourage writing benchmarks from the user perspective.
- I also retired `benchmarks.ARTIFICIAL_DIM_SIZE` because it didn't seem to be adding much, beyond unnecessary complexity.

---
[Consult Iris pull request check list]( https://scitools-iris.readthedocs.io/en/latest/developers_guide/contributing_pull_request_checklist.html)

---
Add any of the below labels to trigger actions on this PR:

- https://github.com/SciTools/iris/labels/benchmark_this

### CODE REVIEW

File: benchmarks/benchmarks/cube.py
Comment (trexfeathers) [inline benchmarks/benchmarks/cube.py:1]: Note that this is being misrepresented by Git/GitHub. The original `cube.py` was moved to `benchmarks/unit_style/cube.py`, while a NEW `benchmarks.cube.py` was written.

File: benchmarks/benchmarks/unit_style/cube.py
Comment (trexfeathers) [inline benchmarks/benchmarks/unit_style/cube.py:1]: https://github.com/SciTools/iris/pull/5949/files#r1596905533

**Review by ESadek-MO — COMMENTED** [review]
Looks good to me, thanks @trexfeathers! Just a couple of potential text changes

File: benchmarks/README.md [hunk 1]
```diff
@@ -62,6 +62,23 @@ interest. Is set during the benchmark runner `cperf` and `sperf` sub-commands.
...
+### What benchmarks to write
+
+It is not possible to maintain a full suite of 'unit style' benchmarks:
+
+* Benchmarks take longer to run than tests.
+* Small benchmarks are more vulnerable to noise - they report a lot of false
+positive regressions.
+
+We therefore recommend writing benchmarks representing scripts or single
+operations that are likely to be run at the user level.
```
[the remark this answers is not in the document]
Comment (trexfeathers) [reply benchmarks/README.md:74]: I went to change it but PyCharm wanted me to change it back so I guess `run` is correct.

File: benchmarks/README.md [hunk 2]
```diff
@@ -62,6 +62,23 @@ interest. Is set during the benchmark runner `cperf` and `sperf` sub-commands.
...
+
+The drawback of this approach: a reported regression is less likely to reveal
+the root cause (e.g. a regression in coordinate creation time observed only
```
[the remark this answers is not in the document]
Comment (trexfeathers) [reply benchmarks/README.md]: ba313f32fe3c07adce221f39e1aff057cb63d24b

Any better?

**Review by ESadek-MO — APPROVED** [review]
I'm happy with this!

**Comment by trexfeathers** [comment]
Thanks @ESadek-MO!

**Pull request merged**